NEW_README.md

```diff
@@ -141,15 +140,155 @@ Run prints its output if either,
+
+#### Optional Fields
+
+##### `description`
+
+A human-readable description of the task. It's displayed in the TUI task list, the output of `run -list`, and can be used to document the task's purpose. It can be one line or multiline.
+
+##### `dependencies`
+Lists other task IDs that should run alongside this task. 
+
+- If task A depends on task B, B starts before A.
+- If B is a "long" task, A will start 500ms after B starts, and if B is a "short" task, A will start as soon as B completes.[^](#type)
+  
+##### `triggers`
+Lists other task IDs that should run alongside this task, and when completed successfully, cause this task to restart. 
+
+- Triggers are similar to dependencies but specifically for re-execution upon successful completion of the trigger task.
+- If task A is triggered by task B, and both A and B are running, a successful completion of B will trigger A to restart.
+- **Triggers are not considered when a task is initially run.** They only affect task restarts.
+- **Long tasks cannot be triggers.** It is invalid to use a long task as a trigger, since long tasks aren't expected to end.
+
+>[!NOTE]
+> Task IDs listed as `dependencies` and `triggers` can cross directory boundaries (e.g., `"css/build"` refers to a task in `./css/tasks.toml`).
+
+##### `watch`
+Defines file paths or globs to monitor for changes. Any detected change triggers a task restart. Examples include:
+- `"."` for the current directory (excluding subdirectories)
+- `"**"` for any change within the working directory
+- Specific paths like `"./src/website/**/*.js"` for targeted file types.
+
+##### `env`
+A map of environment variables provided to the task's execution environment. 
+
+- These variables are available to the command running the task.
+- They are appended to the current environment, overriding any existing variables with the same name.
+- They persist for the duration of the task's execution.
+
+#### `cmd`
+The shell command to execute as the task. It's run in a new bash process:
+
+    $ bash -c "$CMD"
+
+In simple cases, the command can be a single line:
+
+```toml
+[[task]]
+  id = "clean"
+  type = "short"
+  cmd = "go clean -testcache && go clean -modcache"
+```
+
+For more complex commands, or to aid readability, use a multiline string:
+
+```toml
+[[task]]
+  id = "clean"
+  type = "short"
+  cmd = """
+    echo "Cleaning up..."
+    set -x
+
+    rm -rf ./bin
+    go clean -testcache
+    go clean -modcache
+  """
+```
+
 ## Programmatic Use
 
-Run can be used and extended programmatically through its Go API. For more
-information, including a conceptual overview of the architecture, example code,
-and reference documentation, see [the godoc][godoc].
+Run can be used and extended programmatically through its Go API. 
+
+In this small example, we use components from Run to build our own version of the run CLI tool. See cmd/run for the source of the -real- run CLI, which isn't too much more complex. 
+
+```go
```

Comment:
this is nice for users, but not nice for maintainers — it is too easy to break this code without noticing (vs separate examples_xyz.go files, which are covered by go test)

generating the readme, like we do for usage, would be best of both worlds, but that sounds like a lot of work…

maybe now that we have two generated readme components, we ought to have a more robust readme compiler, but that also sounds like a lot of work…